integration/airflow/openlineage/airflow/extractors/python_extractor.py

```diff
@@ -36,6 +37,20 @@ def extract(self) -> Optional[TaskMetadata]:
                         # We're on worker and should have access to DAG files
                         source_code
                     )
+                },
+                run_facets={
+
+                    # The BashOperator is recorded as an "unknownSource" even though we have an
+                    # extractor, as the <i>data lineage</i> cannot be determined from the operator
+                    # directly.
+                    "unknownSourceAttribute": UnknownOperatorAttributeRunFacet(
+                        unknownItems=[
+                            UnknownOperatorInstance(
+                                name="PythonOperator",
+                                properties=self.operator
```

Comment:
I don't think you can put everything here. Imagine `self.operator` that holds `thread` - it will fail on serialization.

Reply:
This has been fixed. JSON now looks like

```json
{
  "unknownSourceAttribute": {
    "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.6.0/integration/airflow",
    "_schemaURL": "https://raw.githubusercontent.com/OpenLineage/OpenLineage/main/spec/OpenLineage.json#/definitions/BaseFacet",
    "unknownItems": [
      {
        "name": "PythonOperator",
        "properties": {
          "_dag": "<<non-serializable: DAG>>",
          "_downstream_task_ids": [],
          "_inlets": {
            "auto": false,
            "datasets": [],
            "task_ids": []
          },
          "_log": "<<non-serializable: Logger>>",
          "_outlets": {
            "datasets": []
          },
          "_upstream_task_ids": [
            "just_wait"
          ],
          "depends_on_past": false,
          "do_xcom_push": true,
          "email": [
            "datascience@example.com"
          ],
          "email_on_failure": false,
          "email_on_retry": false,
          "executor_config": {},
          "inlets": [],
          "op_args": [],
          "op_kwargs": {},
          "outlets": [],
          "owner": "datascience",
          "params": {},
          "pool": "default_pool",
          "pool_slots": 1,
          "priority_weight": 1,
          "provide_context": false,
          "python_callable": "<<non-serializable: function>>",
          "queue": "default",
          "retries": 0,
          "retry_delay": "<<non-serializable: timedelta>>",
          "retry_exponential_backoff": false,
          "start_date": "<<non-serializable: datetime>>",
          "task_id": "emit_event",
          "trigger_rule": "all_success",
          "wait_for_downstream": false,
          "weight_rule": "downstream"
        }
      }
    ]
  }
}
```
Note the non-serializable fields are called out